presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -339,7 +339,7 @@ private void flushStripe(FlushReason flushReason)
         orcDataSink.write(outputData);
 
         // open next stripe
-        columnWriters.forEach(ColumnWriter::reset);
```

Comment:
This may not actually solve the problem because if the original exception is thrown after `columnWriters.forEach(ColumnWriter::close);` in `bufferStripeData` and before this line, then during `rollback`, we will still try to close column writers before resetting them. 

By reading the code, the following part could potentially be the place in `flushStripe` throwing the error causing column writers to be closed.
```java
// write all data
orcDataSink.write(outputData);
```

As the first step, could you write a unit test to verify the stacktrace can be reproduced if the caller calls `orcWriter.write(page)` followed by `orcWriter.close()`? It doesn't have to be the full stackstrace matching. To make sure we can get into closing the column writers twice would be good enough. `OrcDataSink` is an interface that can be mocked.

In terms of the actual solution, I may need to think about it. But feel free to propose a good one.

Reply:
Sure, Ill get started on that unit test.